pkg/cmd/ssh/command.go

```diff
@@ -0,0 +1,460 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ssh
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/target"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	corev1alpha1helper "github.com/gardener/gardener/pkg/apis/core/v1alpha1/helper"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	operationsv1alpha1 "github.com/gardener/gardener/pkg/apis/operations/v1alpha1"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	// SSHBastionUsername is the system username on the bastion host.
+	SSHBastionUsername = "gardener"
+	// SSHNodeUsername is the system username on any of the shoot cluster nodes.
+	SSHNodeUsername = "gardener"
+	// SSHPort is the TCP port on a bastion instance that allows incoming SSH.
+	SSHPort = 22
+)
+
+// NewCommand returns a new ssh command.
+func NewCommand(f util.Factory, o *Options) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "ssh",
+		Short: "Establish an SSH connection to a Shoot cluster's node",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(f, cmd, args, o.IOStreams.Out); err != nil {
+				return fmt.Errorf("failed to complete command options: %w", err)
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			return runCommand(f, o)
+		},
+	}
+
+	cmd.Flags().BoolVar(&o.Interactive, "interactive", o.Interactive, "Open an SSH connection instead of just providing the bastion host.")
```

Comment:
add the information that a node name argument has to be provided (as you described it in the godoc on the `Options` struct for the `Interactive` flag)